### PR TITLE
Support RedHat Enterprise Linux os in the machine controller

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Adding the support for RHEL os in the machine controller.

**Which issue(s) this PR fixes**
Fixes kubermatic/kubermatic#4965

```release-note
NONE
```

### CODE REVIEW

File: pkg/userdata/rhel/provider.go [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+		req.CloudConfig = *pconfig.OverwriteCloudConfig
+	}
+
+	if pconfig.Network != nil {
+		return "", errors.New("static IP config is not supported with RHEL")
+	}
+
+	rhelConfig, err := LoadConfig(pconfig.OperatingSystemSpec)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse OperatingSystemSpec: '%v'", err)
```
Comment (xrstf) [inline pkg/userdata/rhel/provider.go]: Why do we quote the error? (`'%v'`)

File: pkg/userdata/rhel/provider_test.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var (
+	update = flag.Bool("update", false, "update testdata files")
+
+	pemCertificate = `-----BEGIN CERTIFICATE-----
```
Comment (xrstf) [inline pkg/userdata/rhel/provider_test.go]: Is it obvious what this certificate is or should we leave a comment?

Comment (moadqassem) [reply pkg/userdata/rhel/provider_test.go]: This is a certificate which actually used in all our userdata providers testing

File: pkg/userdata/rhel/provider.go (hunk 1 above)
Comment (moadqassem) [reply pkg/userdata/rhel/provider.go]: This is actually a copy and past mistake from this code here: 
https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/centos/provider.go#L48

Will fix it in both centos and rhel

**Pull request merged**